docs/developing/add-new-type.markdown

```diff
@@ -2,168 +2,180 @@
-    // default contract definition for purposes of validation, so you don't need
-    // to duplicate anything that already exists at the top level.
-    // This schema is used for validating new Pokémon cards, and for creating
-    // the form used to create them.
-    "schema": {
-      "type": "object",
-      "properties": {
-        "data": {
-          type: "object",
-          title: "Pokèmon",
-          properties: {
-            Height: {
-              type: "number"
-            },
-            Weight: {
-              type: "number"
-            },
-            Description: {
-              type: "string"
-            },
-            Abilities: {
-              type: "string"
-            },
-            pokedex_number: {
-              title: "National Pokèdex Number",
-              type: "number"
-            },
-            caught: {
-              type: "boolean"
-            },
-            first_seen: {
-              title: "First seen",
-              description: "The first time you saw this pokèmon",
-              type: "string",
-              format: "date-time"
-            }
-          }
-        }
-      }
-    }
-  },
-  "requires": [],
-  "capabilities": []
-}
+## Add a new type definition contract
+
+New type contracts should be added to a [plugin](./plugins.markdown) repo. Currently most contracts are located in the [`contracts` directory](https://github.com/product-os/jellyfish-plugin-default/tree/master/lib/contracts) of the [`jellyfish-plugin-default` plugin](https://github.com/product-os/jellyfish-plugin-default)
+
+A type contract consists of the standard contract fields and a JSON schema that defines the shape of the new type. For example, a new type contract that defines a Pokémon would look like this
+
+
+```typescript
+import type { ContractDefinition } from '@balena/jellyfish-types/build/core';
+
+export const pokemon: ContractDefinition = {
+	// The slug is the machine name of this type and will appear in the "type"
+	// field of all Pokémon contracts derived from this type definition
+	slug: 'pokemon',
+	// Since this is a type definition, it is of the type 'type'. The type is also
+	// versioned, to indicate which version of the 'type' type it will validate
+	// against
+	type: 'type@1.0.0',
+	// The name field is the friendly name used when interacting with the type
+	// definition in UI (e.g. CRUD operations, forms)
+	name: 'Pokémon',
+	// Markers are used for defining permissions and will be explained elsewhere
+	// in the documentation
+	markers: [],
+	// The tag field is used to contain tag data, this is an array of strings
+	tags: [],
+	// In Jellyfish, contracts are "soft deleted" and the active field is used to
+	// indicate whether or not the contract is in this state or not. An active value of
+	// "false" would indicate that the contract has been deleted
+	active: true,
+	// ??
+	requires: [],
+	// ??
+	capabilities: [],
```

Comment:
I think you can drop references to `requires` and `capabilities` as they're not required and not really even used. I've been here for over 2 years and have never used them, I'd be interested if you could find a case in the codebase anywhere that actually uses these.

Reply:
done